### PR TITLE
Allow worldmodel and entity input

### PR DESCRIPTION
**Pull request by PetervDooren**

### CODE REVIEW

File: ed_sensor_integration/tools/segmenter.cpp [hunk 1]
```diff
@@ -182,25 +124,46 @@ int main(int argc, char **argv)
...
     {
-        usage();
-        return 1;
+        if (argc == 2){
+            std::cout << "No worldmodel provided! Using empty worldmodel and no segmentation area" << std::endl;
+        }
+        else{
+            usage();
+            return 1;
+        }
```
Comment (MatthijsBurgh) [inline ed_sensor_integration/tools/segmenter.cpp]:
```suggestion
        if (argc == 2)
        {
            std::cout << "No worldmodel provided! Using empty worldmodel and no segmentation area" << std::endl;
        }
        else
        {
            usage();
            return 1;
        }
```

File: ed_sensor_integration/tools/segmenter.cpp [hunk 2]
```diff
@@ -182,25 +124,46 @@ int main(int argc, char **argv)
...
     {
         std::cerr << "Path '" << path << "' does not exist." << std::endl;
         return 1;
     }
 
+    ed::WorldModel world_model;
+    std::string entity_id;
+    ed::EntityConstPtr e;
+
+    if (argc == 4) {
```
Comment (MatthijsBurgh) [inline ed_sensor_integration/tools/segmenter.cpp]:
```suggestion
    if (argc == 4)
    {
```

File: ed_sensor_integration/tools/segmenter.cpp [hunk 3]
```diff
@@ -182,25 +124,46 @@ int main(int argc, char **argv)
...
+        std::string model_name = argv[2];
+
+
+        if (!loadWorldModel(model_name, world_model)){
```
Comment (MatthijsBurgh) [inline ed_sensor_integration/tools/segmenter.cpp]:
```suggestion
        if (!loadWorldModel(model_name, world_model))
        {
```

File: ed_sensor_integration/tools/segmenter.cpp [hunk 4]
```diff
@@ -240,6 +203,12 @@ int main(int argc, char **argv)
                 i_snapshot = snapshots.size();
                 snapshots.push_back(Snapshot());
                 Snapshot& snapshot = snapshots.back();
+                snapshot.world_model = world_model;
+                if (e)
+                    snapshot.area_description = std::string("on_top_of ").append(entity_id);
+                else
+                    snapshot.area_description = "";
+                std::cout << "area description" << snapshot.area_description << std::endl;
```
Comment (MatthijsBurgh) [inline ed_sensor_integration/tools/segmenter.cpp]:
```suggestion
                snapshot.world_model = world_model;
                if (e)
                    snapshot.area_description = std::string("on_top_of ").append(entity_id);
                else
                    snapshot.area_description = "";

                std::cout << "area description" << snapshot.area_description << std::endl;
```
Better to see what is in the `else` and what not.

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
I expected better of you @PetervDooren after our talk about styleguides 😉

We are changing a current tool. What was the old tool doing and is the new behaviour similar? Or has the tool completely changed? In that case we are actually removing a tool and creating a new one. And we should have a valid reason to drop the old one.

**Comment by PetervDooren** [comment]
The old behaviour of this tool is maintained. Hence the option to provide only 2 arguments instead of 4. As for style guides. We didn't get to that lecture yet :p

File: ed_sensor_integration/tools/segmenter.cpp [hunk 5]
```diff
@@ -182,25 +124,50 @@ int main(int argc, char **argv)
...
+    ed::WorldModel world_model;
+    std::string entity_id;
+    ed::EntityConstPtr e;
+
+    if (argc == 4)
+    {
+        std::string model_name = argv[2];
+
+
+        if (!loadWorldModel(model_name, world_model))
```
Comment (MatthijsBurgh) [inline ed_sensor_integration/tools/segmenter.cpp]:
```suggestion

        if (!loadWorldModel(model_name, world_model))
```

**Comment by MatthijsBurgh** [comment]
In the old code I see `annotations`, but this is nowhere in the new code.

**Comment by PetervDooren** [comment]
closed as these issues were adressed in PR #49

**Pull request closed without merging**